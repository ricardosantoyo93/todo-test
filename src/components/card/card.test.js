import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ShallowRenderer from 'react-test-renderer/shallow';

import { Card } from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Card', () => {
  let wrapper = null;

  const card = {
      title: "",
      date: "",
      content: ""
  }

  beforeEach(() => {
    wrapper = shallow(<Card card={card} />);
  });
  
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Card card={card} />);
    expect(tree).toMatchSnapshot();
  });

  test('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('should have custom content', () => {
    card.content = "Test"
    wrapper = shallow(<Card card={card} />);
    expect(wrapper.contains("Test")).toBe(true);
  });
});
