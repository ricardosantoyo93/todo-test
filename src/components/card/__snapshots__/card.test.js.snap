// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<Card
  actions={
    Array [
      undefined,
      undefined,
      <Button
        className="delBtn"
        floating={true}
        icon={
          <Icon>
            close
          </Icon>
        }
        node="button"
        small={true}
        tooltip="Delete Card"
        waves="light"
      />,
    ]
  }
  className="blue-grey darken-1"
  closeIcon={
    <Icon>
      close
    </Icon>
  }
  revealIcon={
    <Icon>
      more_vert
    </Icon>
  }
  textClassName="white-text"
  title=""
>
  <span
    className="date"
  >
    
  </span>
   
  
</Card>
`;
